File: Assets/Scripts/CfgManager.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using System.IO;

public class CfgManager : MonoBehaviour {

	private static CfgManager instance;
	public static CfgManager GetInstance() {
		GameObject main = GameObject.Find("Main");
		if (main == null) {
			main = new GameObject("Main");
			DontDestroyOnLoad(main);
		}
	
		if (instance == null) {
			instance = main.AddComponent<CfgManager>();
		}
		return instance;
	}

	public string cfg_equips;
	public string cfg_items;
	public string cfg_skills;

	public void Init() {
		Debug.Log("CfgManager:Init");

		cfg_equips = ResourceManager.GetInstance().LoadAsset("cfg/equips").ToString();
		cfg_items = ResourceManager.GetInstance().LoadAsset("cfg/items").ToString();
		cfg_skills = ResourceManager.GetInstance().LoadAsset("cfg/skills").ToString();
	}

}

File: Assets/Lua/CfgManager.lua
--[[
	CfgManager -配置管理
]]

local CfgManager = {}
local this = CfgManager

function CfgManager.Init()
	print ('CfgManager.Init')

	this.items = {}
	this.equips = {}

	local data = resMgr:LoadAsset('Cfg/item'):ToString()
	local obj = json.decode(data)
	this.items = obj
	print ('items count ', #this.items)

	local data = resMgr:LoadAsset('Cfg/equip'):ToString()
	local obj = json.decode(data)
	this.equips = obj
	print ('equips count ', #this.equips)
end

function CfgManager.Items()
	-- body
end


return CfgManager